src/response/response.rs

```diff
@@ -38,14 +46,34 @@ impl Response {
     ///
     #[inline]
     pub fn map(self, map: impl FnOnce(HttpBody<ResponseBody>) -> BoxBody) -> Self {
-        if cfg!(debug_assertions) && matches!(self.inner.body(), HttpBody::Mapped(_)) {
-            // TODO: Replace this with tracing and a proper logger.
-            eprintln!("calling response.map() more than once can create a reference cycle.");
+        if cfg!(debug_assertions) {
+            prefer_unboxed(self.inner.body());
```

Comment:
We can be smarter about this in the future. This works for now.

Reply:
I removed the warning because having 2 pointers can be beneficial in many cases.